static/src/javascripts/projects/commercial/modules/messenger/background.js

```diff
@@ -102,15 +156,23 @@ const setBackground = (specs: AdSpec, adSlot: Node): Promise<any> => {
         }
     };
 
-    return fastdom
-        .write(() => {
-            adSlot.insertBefore(backgroundParent, adSlot.firstChild);
-        })
-        .then(() => {
-            addEventListener(window, 'scroll', onScroll, {
-                passive: true,
-            });
-            onScroll();
+    return getBackground()
+        .then(({ backgroundParent, background }) =>
+            updateStyles(backgroundParent, background)
+        )
+        .then(({ backgroundParent, background }) => {
+            if (!backgroundAlreadyExists) {
+                addEventListener(
```

Comment:
Would it make sense to use the `window:throttledScroll` mediator event here? Or is this not necessary when using passive listeners?

Reply:
My understanding is that, since it is passive, it is fine to use it as-is; it won't hold up the browser doing its thing.